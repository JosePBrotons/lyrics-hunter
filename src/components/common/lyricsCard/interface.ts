export interface ILyricsCardProps {
    artist: string;
    song: string;
    lyrics: string;
    onPress?: () => void;
}