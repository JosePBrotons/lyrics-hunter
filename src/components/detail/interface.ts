export interface IDetailProps {
    route: IRoute;
    navigation: INavigation;
}

export interface INavigation {
    setOptions: (arg0: IOptions) => void;
}

export interface IRoute {
    params: IRouteParams;
}

export interface IRouteParams {
    artist: string;
    song: string;
    lyrics: string;
}

export interface IOptions {
    title: string;
}
