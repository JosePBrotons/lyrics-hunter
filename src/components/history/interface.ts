export interface IHistoryProps {
    navigation: any;
}
