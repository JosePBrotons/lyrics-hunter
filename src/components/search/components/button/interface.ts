export interface IButtonProps {
    text: string;
    disabled: boolean;
}