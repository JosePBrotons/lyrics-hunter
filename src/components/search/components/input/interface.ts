import { ColorValue } from 'react-native';

export interface IInputProps {
    placeholder: string;
    value: string;
    placeholderTextColor: ColorValue;
    onChangeText: (value: React.SetStateAction<string>) => void;
    secureTextEntry?: boolean;
    iconName?: string;
}
