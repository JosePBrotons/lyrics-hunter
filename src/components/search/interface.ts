export interface ISearchProps {
    navigation: any;
}

export interface ILatestSearch {
    artist: string;
    song: string;
    lyrics: string;
}
