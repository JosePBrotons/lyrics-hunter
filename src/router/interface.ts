import { RouteProp } from '@react-navigation/native';

export interface IOptions {
    title: string;
    headerTitleStyle?: IHeaderTitleStyle;
}

export interface IStack {
    name: string;
    component: (props: any) => JSX.Element;
    options?: IOptions;
}

export interface ITabIconProps {
    focused: boolean;
    color: string;
    size: number;
}

export interface IHeaderTitleStyle {
    fontFamily: string;
}

export interface IScreenOptions {
    route: RouteProp<Record<string, object | undefined>, string>;
    navigation: any;
}

export interface ITab {
    name: string;
    component: any
    options?: IOptions;
}