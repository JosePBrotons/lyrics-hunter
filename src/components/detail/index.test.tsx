import React from 'react';
import renderer from 'react-test-renderer';
import Detail from '.';
import { IDetailProps } from './interface';

const props: IDetailProps = {
    navigation: { setOptions: jest.fn },
    route: {
        params: {
            artist: 'Coldplay',
            song: 'Viva La Vida',
            lyrics: 'I used to rule the world...',
        },
    },
};

test('renders correctly', () => {
    const searchCmp = renderer.create(<Detail {...props} />).toJSON();
    expect(searchCmp).toMatchSnapshot();
});
