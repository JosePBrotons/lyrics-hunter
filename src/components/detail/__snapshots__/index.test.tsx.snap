// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "backgroundColor": "rgba(233, 233, 233, 1)",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(69, 69, 69, 1)",
            "fontFamily": "Nunito-Bold",
            "fontSize": 22,
            "lineHeight": 30,
            "marginVertical": 20,
            "textAlign": "center",
          }
        }
      >
        [missing "en.lyricsDetail.artist" translation]: Coldplay
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(106, 106, 106, 1)",
            "fontFamily": "Nunito-Regular",
            "fontSize": 16,
            "lineHeight": 22,
            "textAlign": "center",
          }
        }
      >
        I used to rule the world...
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
